flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/MockExecutionGraphBase.java

```diff
@@ -0,0 +1,424 @@
+        throw new UnsupportedOperationException();
+    }
+
+    @Nullable
+    @Override
+    public CheckpointCoordinator getCheckpointCoordinator() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public KvStateLocationRegistry getKvStateLocationRegistry() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void setJsonPlan(String jsonPlan) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Either<SerializedValue<JobInformation>, PermanentBlobKey> getJobInformationOrBlobKey() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Configuration getJobConfiguration() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public ClassLoader getUserClassLoader() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Throwable getFailureCause() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Iterable<ExecutionJobVertex> getVerticesTopologically() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Iterable<ExecutionVertex> getAllExecutionVertices() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public ExecutionJobVertex getJobVertex(JobVertexID id) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Map<JobVertexID, ExecutionJobVertex> getAllVertices() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public long getNumberOfRestarts() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public int getTotalNumberOfVertices() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Map<IntermediateDataSetID, IntermediateResult> getAllIntermediateResults() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BlobWriter getBlobWriter() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Executor getFutureExecutor() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Map<String, OptionalFailure<Accumulator<?, ?>>> aggregateUserAccumulators() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void setInternalTaskFailuresListener(
+            InternalFailuresListener internalTaskFailuresListener) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void attachJobGraph(List<JobVertex> topologiallySorted) throws JobException {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void transitionToRunning() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void cancel() {
+        throw new UnsupportedOperationException();
```

Comment:
Throwing `UnsupportedOperationExceptions` whenever any method that is not explicitly mocked by the test is called will make this quite annoying to use.

Reply:
For my use-case, I'd almost say the opposite is true. I consider it useful to see right away what I've missed to implement in my mock by having an exception thrown. Making cancel a no-op would require me to first understand why the cancellation hasn't happened as expected. With an exception, I notice it right away.

The MockExecutionGraphBase is really just a utility for the state tests. I agree that it might be of debatable use for others. Would moving the MockBase to the declarative scheduler tests package resolve your concern?

Reply:
After working on fixing the failing tests, I realize that I need to revisit my MockExecutionGraphBase strategy.
I'll move it to the declarative scheduler package and customize it more for my needs.

I'll ping you when I need another review.

Reply:
For reference, [this](https://github.com/apache/flink/pull/14963/files#r578791931) is exactly the kind of inconvenience that I' concerned about.
I was about the suggest a compromise where certain usually non-interesting methods would be mocked in a non-failing way, but you already had that idea earlier it appears 👍